packages/esm-patient-chart-app/src/encounter-list/components/encounter-list.component.tsx

```diff
@@ -0,0 +1,300 @@
+  };
+  filter?: (encounter: any) => boolean;
+  afterFormSaveAction?: () => void;
+}
+
+export const EncounterList: React.FC<EncounterListProps> = ({
+  patientUuid,
+  encounterType,
+  columns,
+  headerTitle,
+  description,
+  formList,
+  filter,
+  launchOptions,
+  afterFormSaveAction,
+}) => {
+  const { t } = useTranslation();
+  const { isDead } = usePatientDeathStatus(patientUuid);
+  const formUuids = useMemo(() => formList.map((form) => form.uuid), [formList]);
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+
+  const { formsJson, isLoading: isLoadingFormsJson } = useFormsJson(formUuids);
+
+  const { encounters, total, isLoading, onFormSave, mutate } = useEncounterRows(
+    patientUuid,
+    encounterType,
+    filter,
+    afterFormSaveAction,
+    pageSize,
+    currentPage,
+  );
+
+  const { workspaceWindowSize, displayText, hideFormLauncher } = launchOptions;
+
+  const defaultActions = useMemo(
+    () => [
+      {
+        label: t('viewEncounter', 'View'),
+        form: {
+          name: formsJson[0]?.name,
+        },
+        mode: 'view',
+        intent: '*',
+      },
+      {
+        label: t('editEncounter', 'Edit'),
+        form: {
+          name: formsJson[0]?.name,
+        },
+        mode: 'edit',
+        intent: '*',
+      },
+      {
+        label: t('deleteEncounter', 'Delete'),
+        form: {
+          name: formsJson[0]?.name,
+        },
+        mode: 'delete',
+        intent: '*',
+      },
+    ],
+    [formsJson, t],
+  );
+
+  const createLaunchFormAction = (encounter, mode) => () =>
+    launchEncounterForm(formsJson[0], mode, onFormSave, null, encounter.uuid, null, workspaceWindowSize, patientUuid);
+
+  const handleDeleteEncounter = useCallback(
+    (encounterUuid, encounterTypeName) => {
+      const close = showModal('delete-encounter-modal', {
+        close: () => close(),
+        encounterTypeName: encounterTypeName || '',
+        onConfirmation: () => {
+          const abortController = new AbortController();
+          deleteEncounter(encounterUuid, abortController)
+            .then(() => {
+              onFormSave();
+              showSnackbar({
+                isLowContrast: true,
+                title: t('encounterDeleted', 'Encounter deleted'),
+                subtitle: `Encounter ${t('successfullyDeleted', 'successfully deleted')}`,
+                kind: 'success',
+              });
+            })
+            .catch(() => {
+              showSnackbar({
+                isLowContrast: false,
+                title: t('error', 'Error'),
+                subtitle: `Encounter ${t('failedDeleting', "couldn't be deleted")}`,
+                kind: 'error',
+              });
+            })
+            .finally(() => {
+              mutate();
+              close();
+            });
+        },
+      });
+    },
+    [onFormSave, t, mutate],
+  );
+
+  const tableRows = encounters.map((encounter) => {
```

Comment:
Ideally, this would be a separate component (in this same file) or, at the very least, wrapped in a `useMemo()`.